lib/string_reverse.rb

```diff
@@ -1,6 +1,17 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
+
 def string_reverse(my_string)
-  raise NotImplementedError
+  return nil if my_string == nil
+
+  length = my_string.length
+
+  (length / 2).times do |i|

```

Comment:
Tiny personal-preference nit: `i` is a pretty standard variable name for things like these, but using something more explanatory (e.g. `start_index`) can make your code _more explicit_ and in turn easier for some to read. (Of course, others find _more concise_ code more readable - so this is _perfectly fine_ if you're in the latter camp, as are most Ruby developers. 🙂)

(However - more explicit code can be _very_ helpful in interviews, because it reduces the chance that you or your interviewer will accidentally mess up variable names. 🙂)